integration/spark/src/main/common/java/io/openlineage/spark/agent/util/PlanUtils.java

```diff
@@ -183,4 +189,28 @@ public static Path getDirectoryPath(Path p, Configuration hadoopConf) {
       return p;
     }
   }
+
+  public static List<Path> findRDDPaths(List<RDD<?>> fileRdds) {
+    return fileRdds.stream()
+        .flatMap(
+            rdd -> {
+              if (rdd instanceof HadoopRDD) {
+                HadoopRDD hadoopRDD = (HadoopRDD) rdd;
+                Path[] inputPaths = FileInputFormat.getInputPaths(hadoopRDD.getJobConf());
+                Configuration hadoopConf = hadoopRDD.getConf();
+                return Arrays.stream(inputPaths)
+                    .map(p -> PlanUtils.getDirectoryPath(p, hadoopConf));
+              } else if (rdd instanceof FileScanRDD) {
+                FileScanRDD fileScanRDD = (FileScanRDD) rdd;
+                return ScalaConversionUtils.fromSeq(fileScanRDD.filePartitions()).stream()
+                    .flatMap(fp -> Arrays.stream(fp.files()))
+                    .map(f -> new Path(f.filePath()).getParent())
+                    .filter(Objects::nonNull);
```

Comment:
You're right. In case something goes wrong with new Path, exception is thrown.

I reused this code from here:
https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/AbstractRDDNodeVisitor.java#L58
to make a common method for both usages.